var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -157,12 +157,15 @@ class Petsc(Package, CudaPackage, ROCmPackage):
     variant("kokkos", default=False, description="Activates support for kokkos and kokkos-kernels")
     variant("fortran", default=True, description="Activates fortran support")
 
-    # https://github.com/spack/spack/issues/37416
-    conflicts("^rocprim@5.3.0:5.3.2", when="+rocm")
-    # petsc 3.20 has workaround for breaking change in hipsparseSpSV_solve api,
-    # but it seems to misdetect hipsparse@5.6.1 as 5.6.0, so the workaround
-    # only makes things worse
-    conflicts("^hipsparse@5.6", when="+rocm @3.20.0")
+    with when("+rocm"):
+        # https://github.com/spack/spack/issues/37416
+        conflicts("^rocprim@5.3.0:5.3.2")
+        # hipsparse@5.6.0 broke hipsparseSpSV_solve() API, reverted in 5.6.1.
+        conflicts("^hipsparse@5.6.0", when="@:3.19")
+        # petsc@3.20 has workarounds for 5.6.0, but it incorrectly detects
+        # hipsolver 5.6.1 as 5.6.0, so now it works with the broken 5.6.0 but
+        # not with the fixed 5.6.1 release.
+        conflicts("^hipsparse@5.6.1", when="@3.20.0")
```

Comment:
with the fix likely in 3.20.1 - updating the suggested change

```suggestion
        # petsc@3.19 works with 5.6.0, but not 5.6.1+
        # petsc@3.20.0 works with 5.6.0, 5.7.0 - but not 5.6.1 [its detected as 5.6.0]
        # petsc@3.20.1 has a fix - so should work with 5.6.0, 5.6.1, 5.7.0 
        conflicts("^hipsparse@5.6.1", when="@3.20.0")
        conflicts("^hipsparse@5.6.1:", when="@3.19")
        conflicts("^hipsparse@5.6:", when="@:3.18")
```

Reply:
I don't care too much, but since `hipsparse@5.6.1` fixed the breaking change, `petsc@:3.18` should be compatible with 5.6.1 (but not 5.6.0), right?